staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/errors.go

```diff
@@ -48,8 +48,10 @@ func Forbidden(attributes authorizer.Attributes, w http.ResponseWriter, req *htt
 
 func forbiddenMessage(attributes authorizer.Attributes) string {
 	username := ""
+	groups := []string{}
```

Comment:
This list could be really long.  The user helps differentiate cases where a single process has multiple clients (`kube-controller-manager` as an example).  I don't think groups are generally needed.